### PR TITLE
[Feature] [55628] Init Project

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- [#Task 55628: Tạo PR Initial Project](https://edu-redmine.sun-asterisk.vn/issues/55628)
## WHAT
Setup project
1.Add .gitignore in root
- Create file .gitignore in root
- Copy content of file into recent created file
2. Install Swiftlint
Install
3. Add pull request template
- Create file .github/pull_request_template.md
- Copy content of file into recent created file.
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/Zts4P8 |<li>- [ ] yes</li>|<li>- [ ] yes</li>
Message | Sun* commit message | Template: [TASK/FREATURE/BUG/SUPPORT][RedmineID] + message |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: MovieDB/ViewController.swift [hunk 1]
```diff
@@ -0,0 +1,16 @@
+//
+//  ViewController.swift
+//  MovieDB
+//
+//  Created by le.n.t.trung on 03/11/2022.
+//
+
+import UIKit
+
+class ViewController: UIViewController {
```
Comment (tunghs-1253) [inline MovieDB/ViewController.swift]:
```suggestion
final class ViewController: UIViewController {
```

File: MovieDB/AppDelegate.swift [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                     didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        // Override point for customization after application launch.
+        return true
+    }
+
+    // MARK: - Core Data stack
+
+    lazy var persistentContainer: NSPersistentContainer = {
+        /*
+         The persistent container for the application. This implementation
```
Comment (tunghs-1253) [inline MovieDB/AppDelegate.swift]: Mấy comment kiểu này nên xoá trông cho gọn nha Trung :v

File: MovieDB/AppDelegate.swift [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        let context = persistentContainer.viewContext
+        if context.hasChanges {
+            do {
+                try context.save()
+            } catch {
+                let nserror = error as NSError
+                fatalError("Unresolved error \(nserror), \(nserror.userInfo)")
+            }
+        }
+    }
```
Comment (ghost) [inline MovieDB/AppDelegate.swift]: tách thành 1 file riêng e

Comment (ghost) [reply MovieDB/AppDelegate.swift]: Dạ

**Review by tunghs-1253 — APPROVED** [review]

File: MovieDB/Manager/CoreDataManager.swift [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+//  CoreDataManager.swift
+//  MovieDB
+//
+//  Created by le.n.t.trung on 03/11/2022.
+//
+
+import Foundation
+import CoreData
+
+public class PersistentContainer {
```
Comment (ghost) [inline MovieDB/Manager/CoreDataManager.swift]:
```suggestion
final class PersistentContainer {
```

File: MovieDB/Manager/CoreDataManager.swift [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        let context = persistentContainer.viewContext
+        if context.hasChanges {
+            do {
+                try context.save()
+            } catch {
+                let nserror = error as NSError
+                fatalError("Unresolved error \(nserror), \(nserror.userInfo)")
+            }
+        }
+    }
```
Comment (ghost) [inline MovieDB/Manager/CoreDataManager.swift]: bỏ static hết đi e, dùng singleton

File: MovieDB/Manager/CoreDataManager.swift [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+    static var persistentContainer: NSPersistentContainer = {
+        let container = NSPersistentContainer(name: "Model")
+        container.loadPersistentStores(completionHandler: { (_, error) in
+            if let error = error as NSError? {
+                fatalError("Unresolved error \(error), \(error.userInfo)")
+            }
+        })
+        return container
+    }()
```
Comment (ghost) [inline MovieDB/Manager/CoreDataManager.swift]: thằng `persistentContainer` nên khởi tạo ở trong init
```suggestion
    static let shared =  PersistentContainer()
    private var persistentContainer: NSPersistentContainer?
    
    private init() {
        peristentContainer = NSPersistentContainer(name: "Model")
        peristentContainer.loadPersistentStores(completionHandler: { (_, error) in
            if let error = error as NSError? {
                fatalError("Unresolved error \(error), \(error.userInfo)")
            }
        })
    }
```

**Review by ghost — APPROVED** [review]

**Pull request merged**